hugo/content/docs/smodel.md

```diff
@@ -0,0 +1,554 @@
+SLabelImpl --> SShapeElementImpl
+SCompartmentImpl --> SShapeElementImpl
+SGraphIndex --> ModelIndexImpl
+{{< /mermaid>}}
+
+## Features
+
+Features can be viewed as a set of functionalities that a given element can support. They can enrich the behavior and usability of the diagram model. SModelElements can have default features that will apply to all model elements that are instance of this SModelElement.
+
+Features can be added in a inherited class by setting the `DEFAULT_FEATURES` field. This way, the default features will be enabled for all instances of the class.
+
+Let's have a look at the `SNodeImpl` class and how it implements default features:
+
+```typescript
+export class SNodeImpl extends SConnectableElementImpl implements Selectable, Fadeable, Hoverable {
+    static readonly DEFAULT_FEATURES = [connectableFeature, deletableFeature, selectFeature, boundsFeature,
+        moveFeature, layoutContainerFeature, fadeFeature, hoverFeedbackFeature, popupFeature];
+    ...
+}
+```
+
+It is possible to fine-tune the behavior in the [dependency injection container](../dependency_injection/#the-container) by enabling or disabling features for a given model element type.
+
+```typescript
+configureElement('my-node-type', SNodeImpl, RectangularNodeView, {enable: [layoutableChildFeature], disable: [moveFeature]})
+```
+
+To ensure that the required additional properties are present on the model element, Sprotty comes with a set of interfaces known as `SModelExtension`.
+
+### alignFeature
+
+Controls the position adjustment of an element. It is given by the `alignment: Point` property of an element.
+It applies a `translate` to the svg element.
+
+*Interface:* `Alignable`
+
+```typescript
+export interface Alignable extends SModelExtension {
+    alignment: Point
+}
+```
+
+### boundsFeature
+
+Controls if the bounds of an element can be calculated and updated. The element needs a `position` and `size`.
+
+*Interface:* `BoundsAware`
+
+```typescript
+export interface BoundsAware extends SModelExtension{
+    bounds: Bounds
+}
+```
+
+### connectableFeature
+
+Controls if an element can be connected to other elements. To be connectable, the element must be an `SRoutableElementImpl` and have a role of source or target.
+
+*Interface:* `Connectable`
+
+```typescript
+
+export interface Connectable extends SModelExtension{
+    canConnect(routable: SRoutableElementImpl, role: 'source' | 'target'): boolean
+}
+```
+
+### creatingOnDragFeature
+
+Controls if an element is created on drag. The element needs to implement a `createAction` method that returns an `Action` that creates the element.
+
+*Interface:* `CreatingOnDrag`
+
+```typescript
+export interface CreatingOnDrag extends SModelExtension {
+    createAction(id: string): Action
+}
+```
+
+### decorationFeature
+
+Controls if an element is a decoration.
+
+*Interface*: `Decoration`
+
+```typescript
+export interface Decoration extends SModelExtension {}
```

Comment:
I was wondering about this extension. Typescript is structurally-typed, not nominally. If you do not add attributes the `Decoration` type is the same as `SModelExtension`.
Same for `Deletable`.

Reply:
That's how they are in the source code. With some interfaces, they do not add any property. The `SModelExtension` is also an empty interface.

Seems like it was done in a systematic way even for empty interfaces.

Reply:
OK, but what is the purpose of a decoration? Where is it used?